apps/src/music/views/ExemplarPlayerView.tsx

```diff
@@ -0,0 +1,152 @@
+import FontAwesomeV6Icon from '@code-dot-org/component-library/fontAwesomeV6Icon';
+import classNames from 'classnames';
+import React, {useCallback, useEffect, useRef, useState} from 'react';
+
+import {useAppDispatch, useAppSelector} from '@cdo/apps/util/reduxHooks';
+
+import {Source} from '../../lab2/types';
+import MusicBlocklyWorkspace from '../blockly/MusicBlocklyWorkspace';
+import {BlockMode, DEFAULT_PACK} from '../constants';
+import MusicLibrary from '../player/MusicLibrary';
+import MusicPlayer from '../player/MusicPlayer';
+import Simple2Sequencer from '../player/sequencer/Simple2Sequencer';
+import {setPlayerContext, setExemplarPlaybackEvents} from '../redux/musicRedux';
+
+import moduleStyles from './ExemplarPlayer.module.scss';
+
+interface ExemplarPlayerViewProps {
+  source: Source;
+  title: string;
+  labSetPlaying: (playing: boolean) => void;
+}
+
+const ExemplarPlayerView: React.FunctionComponent<ExemplarPlayerViewProps> = ({
+  source,
+  title,
+  labSetPlaying,
+}) => {
+  const dispatch = useAppDispatch();
+  const playerContext = useAppSelector(state => state.music.playerContext);
+
+  const playerRef = useRef<MusicPlayer | null>(null);
+  if (playerRef.current === null) {
+    playerRef.current = new MusicPlayer();
+  }
+  const workspaceRef = useRef<MusicBlocklyWorkspace>(
+    new MusicBlocklyWorkspace()
+  );
+  const simple2SequencerRef = useRef<Simple2Sequencer>(new Simple2Sequencer());
+  const [isPlaying, setIsPlaying] = useState<boolean>(false);
+  // Immediately load the library and source, then compile the song.
+  const onMount = useCallback(async () => {
+    workspaceRef.current.initHeadless();
+
+    workspaceRef.current.loadCode(source);
+    workspaceRef.current.compileSong(
+      {Sequencer: simple2SequencerRef.current},
+      BlockMode.SIMPLE2
```

Comment:
Should this be the level's block mode?

Reply:
In light of the decision memo, I opted to just support Simple2 for now.